clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1176,8 +1176,13 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
         A->getFilteredAttributeNameValuePairs(CGM.getContext());
 
     llvm::AttrBuilder FnAttrBuilder(Fn->getContext());
-    for (const auto &NameValuePair : NameValuePairs)
+    for (const auto &NameValuePair : NameValuePairs) {
+      if (NameValuePair.first == "sycl-floating-point-control" &&
```

Comment:
We shouldn't emit diagnostics in CodeGen. I'm actually not sure if/when community finds CodeGen diagnostic acceptable. Maybe @AaronBallman will have insights. 

Having said that, we don't introspect properties currently in Sema (and we should aim to keep it that way to ensure compile time properties don't just become another version of attributes). So I don't think this diagnostic can be implemented in the FE.

Reply:
Ok, reverted frontend changes.
Changed this to static assert in SYCL headers - error is emitted if introduced compile time property is applied on non-ESIMD kernel. One downside of this is that error will not be generated if to compile with -fsycl-device-only, i.e. error is generated if to compile with -fsycl (this is because Kernel::isESIMD() is specialized in integration header and this specialization is available only for host compilation).  I wasn't able to overcome that and didn't find another way which will not look ugly.
@gmlueck  Would such static assert be an acceptable approach?

Reply:
Yes, that's OK.